mkosi/distributions/postmarketos.py

```diff
@@ -0,0 +1,121 @@
+from mkosi.installer.apk import Apk, ApkRepository
+from mkosi.log import complete_step, die
+from mkosi.run import exists_in_sandbox
+from mkosi.tree import copy_tree
+
+
+class Installer(DistributionInstaller):
+    @classmethod
+    def pretty_name(cls) -> str:
+        return "postmarketOS"
+
+    @classmethod
+    def filesystem(cls) -> str:
+        return "ext4"
+
+    @classmethod
+    def package_type(cls) -> PackageType:
+        return PackageType.apk
+
+    @classmethod
+    def default_release(cls) -> str:
+        return "edge"
+
+    @classmethod
+    def default_tools_tree_distribution(cls) -> Distribution:
+        return Distribution.postmarketos
+
+    @classmethod
+    def package_manager(cls, config: Config) -> type[PackageManager]:
+        return Apk
+
+    @classmethod
+    def setup(cls, context: Context) -> None:
+        Apk.setup(context, list(cls.repositories(context)))
+
+        # Create merged /usr manually for now until our upstream (Alpine Linux) supports it:
+        # https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/85504
+        for dir in ["lib", "bin", "sbin"]:
+            (context.root / "usr" / dir).mkdir(parents=True, exist_ok=True)
+            (context.root / dir).symlink_to(f"usr/{dir}")
+
+        with complete_step("Setting up postmarketOS keyring"):
+            # Create keys directory in sandbox
+            keys_dir = context.sandbox_tree / "etc/apk/keys"
+            keys_dir.mkdir(parents=True, exist_ok=True)
+
+            # Copy keys from various sources (if they exist)
+            tools_usr = context.config.tools() / "usr"
+            source_dirs = [
+                tools_usr / "lib/apk/keys",
+                tools_usr / "share/distribution-gpg-keys/alpine-linux",
+                tools_usr / "share/distribution-gpg-keys/postmarketos",
+            ]
+
+            for source_dir in source_dirs:
+                if source_dir.exists():
+                    copy_tree(
+                        source_dir,
+                        keys_dir,
+                        sandbox=context.sandbox,
+                    )
+
+    @classmethod
+    def install(cls, context: Context) -> None:
+        Apk.install(
+            context,
+            [
+                "postmarketos-baselayout",
+                "postmarketos-release",
+            ],
+            apivfs=False,
+        )
+
+    @classmethod
+    def repositories(cls, context: Context) -> Iterable[ApkRepository]:
+        if context.config.release != "edge":
+            die(f"Only 'edge' release is currently supported, got '{context.config.release}'")
+
+        if context.config.local_mirror:
+            yield ApkRepository(url=context.config.local_mirror)
+            return
+
+        # Alpine repos
+        alpine_mirror = "https://dl-cdn.alpinelinux.org/alpine"
+        for repo_name in ["main", "community", "testing"]:
+            url = f"{alpine_mirror}/{context.config.release}/{repo_name}"
```

Comment:
I think all of these `url` variables can just be inlined, no point in separate variables